core/process.py

```
@@ -0,0 +1,122 @@
+            time.sleep(0.1)
+
+        # Process didn't exit gracefully, force kill it
+        process.kill()
+        process.wait(1.0)  # Wait up to 1 second for it to be killed
+        return False
+    except Exception as e:
+        logger.error(f"Error terminating process: {e}")
+        # Try one last time with kill
+        try:
+            process.kill()
+        except:
```

Ruff (0.8.2)

Do not use bare `except`

(E722)